codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/XmlBindingTraitParserGenerator.kt

```diff
@@ -346,7 +346,7 @@ class XmlBindingTraitParserGenerator(
                 *codegenScope, "Shape" to symbol
             ) {
                 val members = shape.members()
-                rustTemplate("let mut base: Option<#{Shape}> = None;", *codegenScope)
+                rustTemplate("let mut base: Option<#{Shape}> = None;", *codegenScope, "Shape" to symbol)
```

Comment:
Wait, how did this ever work?

Reply:
because rustBlockTemplate actually applies the codegen scope to the entire block, but now it fails because of the new assert